src/util/sighash.rs

```diff
@@ -128,6 +127,71 @@ pub enum SchnorrSigHashType {
     /// Reserved for future use, `#[non_exhaustive]` is not available with current MSRV
     Reserved = 0xFF,
 }
+serde_string_impl!(SchnorrSigHashType, "a SchnorrSigHashType data");
+
+impl SchnorrSigHashType {
+    /// Break the sighash flag into the "real" sighash flag and the ANYONECANPAY boolean
+    pub(crate) fn split_anyonecanpay_flag(self) -> (SchnorrSigHashType, bool) {
```

Comment:
Nit: I think this may be useful outside of the scope of just this library (for instance for some wallet applications)

Reply:
Fair point but unrelated to this PR so leaving as is. If we want this to be public we should do it as an RC fix though, I would not be confident to write the supporting PR message championing the idea other than to copy what you have said verbatim. Can I leave it with you?